### PR TITLE
OBU OTA Server

### PR DESCRIPTION
**Pull request by Michael7371**
# PR Details

## Description

This PR includes the addition of an OBU OTA server that can be deployed in K8 environments or on a local VM to server firmware files for OBU devices to request. Currently, we have only added support for Commsignia OBU's but can be altered to support other vendors.

## How Has This Been Tested?

I have added unit tests to cover most of the functional use cases. This is also deployed in the CDOT GCP environment and has been successfully tested with an OBU device.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ x ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ x ] My changes require new environment variables:
  - [ x ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [ x ] My changes require updates to the documentation:
  - [ x ] I have updated the documentation accordingly.
- [ x ] My changes require updates and/or additions to the unit tests:
  - [ x ] I have modified/added tests to cover my changes.
- [ x ] All existing tests pass.

### CODE REVIEW

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]
There are some merge conflicts to address

**Comment by Michael7371** [comment]
> There are some merge conflicts to address

Should be all good now, let me know if you see any issues with how I managed the merge conflicts. I made sure the unit tests still passed.

File: services/addons/images/obu_ota_server/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+Available REST endpoints:
+
+- / [ **GET** ]
+  - Used as a health check path for K8 deployment. This path is not secured with basic authentication
+- /firmwares/commsignia [ **GET** ]
+  - Secured with basic authentication.
+  - Firmware file name must comply with the following regex naming filter:
+    `(?P<variant>(?P<type>ob4|rs4)-generic(?:-.*)?)-(?:(?P<writable>rw)|ro)(?P<secure>-secureboot)?-(?P<release>.*).tar.sig`
+  - Will either query GCS or the specified local path for a latest list of firmware files. Generates a manifest that includes details about each firmware file along with a `href` field that the OBU uses to get generate the download request. Here is a sample response:
```
Comment (dmccoystephenson) [inline services/addons/images/obu_ota_server/README.md]: > the OBU uses to get generate the download request

Should this be changed to "the OBU uses to generate the download request"?

File: services/addons/tests/firmware_manager/test_download_blob.py [hunk 2]
```diff
@@ -6,51 +6,6 @@
 from addons.images.firmware_manager.download_blob import UnsupportedFileTypeException
```
Comment (dmccoystephenson) [inline services/addons/tests/firmware_manager/test_download_blob.py]: This class was removed but the import statement remains; running into an error when running unit tests due to this

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]

File: services/addons/tests/firmware_manager/test_download_blob.py (hunk 2 above)
Comment (Michael7371) [reply services/addons/tests/firmware_manager/test_download_blob.py]: My mistake, for some reason my vscode wasn't picking up that failing unit test.

Comment (Michael7371) [reply services/addons/tests/firmware_manager/test_download_blob.py]: Fixed now!

File: services/addons/images/obu_ota_server/README.md (hunk 1 above)
Comment (Michael7371) [reply services/addons/images/obu_ota_server/README.md]: Fixed!

**Review by dmccoystephenson — APPROVED** [review]
Code looks good & the unit tests pass!

**Review by drewjj — APPROVED** [review]
This looks great! All tests pass and it is running well in GCP dev.

**Review by mwodahl — APPROVED** [review]
This all looks good to me!

**Pull request merged**